### PR TITLE
S100

### LINKED ISSUE #96: Validation (2017): S100 (Activity Year)
**Edit #:** S100
**Type:** Syntactical
**Category:** Transmittal Sheet (only)
**Column:** Activity Year

Activity year must = year being processed (= 2017).

Additional resources:
* [FFIEC's 2016 HMDA File Spec](https://www.ffiec.gov/hmda/pdf/spec2016.pdf)
* [FFIEC's 2016 HMDA Edits](https://www.ffiec.gov/hmda/pdf/edit2016.pdf)
* [CFPB's 2017 HMDA File Spec](http://www.consumerfinance.gov/hmda/static/for-filers/2017/2017-HMDA-File-Specifications.pdf)

### LINKED ISSUE #563: add year to ValidationContext
Add filing year to `ValidationContext`, now that it exists.  That way, edits that need it can get it that way.

We could perhaps also put the "timestamp" needed for S013 in there.  That one is puzzling because it's still not clear (at least to me) what it means, but the meaning is probably separate from the question of how to pass the information into the edit.  In any case, if it's blocked, then it doesn't need to be part of this ticket.  But if it's easy to go ahead and include, then we might as well.

### PR DESCRIPTION
**Pull request by nickgrippin**
Closes #96 
Closes #563

### CODE REVIEW

**Comment by jmarin** [comment]
LGTM

File: validation/src/test/scala/hmda/validation/rules/ts/syntactical/S100Spec.scala [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import hmda.model.institution.Agency.CFPB
+import hmda.model.institution.ExternalIdType.{ FdicCertNo, FederalTaxId, RssdId }
+import hmda.model.institution.InstitutionStatus.Active
+import hmda.model.institution.InstitutionType.Bank
+import hmda.model.institution.{ ExternalId, Institution }
+import hmda.validation.context.ValidationContext
+import hmda.validation.dsl.{ Failure, Success }
+import org.scalatest.{ MustMatchers, WordSpec }
+
+class S100Spec extends WordSpec with MustMatchers {
```
Comment (kgudel) [inline validation/src/test/scala/hmda/validation/rules/ts/syntactical/S100Spec.scala]: I think it might be useful to also include a test for: `ValidationContext(None, None)`

Comment (nickgrippin) [inline validation/src/test/scala/hmda/validation/rules/ts/syntactical/S100Spec.scala]: Good suggestion!  It should be added it now.

Comment (kgudel) [inline validation/src/test/scala/hmda/validation/rules/ts/syntactical/S100Spec.scala]: Thanks! I think it helps document the behavior a bit better :)
